src/sensors/CyclePowerData.cpp

```diff
@@ -10,20 +10,15 @@
 
 bool CyclePowerData::hasHeartRate() { return false; }
 
-bool CyclePowerData::hasCadence() { return this->cadence > 0 || this->missedReadingCount < 3; }
+bool CyclePowerData::hasCadence() { return !isnan(this->cadence); }
 
-bool CyclePowerData::hasPower() { return this->power != INT_MIN; }
+bool CyclePowerData::hasPower() { return true; }
 
 bool CyclePowerData::hasSpeed() { return false; }
 
 int CyclePowerData::getHeartRate() { return INT_MIN; }
 
-float CyclePowerData::getCadence() {
-  if (!this->hasCadence()) {
-    return NAN;
-  }
-  return this->cadence;
-}
+float CyclePowerData::getCadence() { return this->cadence; }
```

Comment:
This seems to be fine because we have a requirement that you need to call the `has*` guard before accessing the data.